File: src/chainOfResponsibility/python/RoleCheckLink.py
from src.chainOfResponsibility.python.ChainLink import ChainLink


class RoleCheckLink(ChainLink):
    def check(self, username: str, password: str, token: str) -> bool:
        if username == "admin":
            print("admin")
            return True

        print("user")
        return self.next(username, password, token)

File: src/chainOfResponsibility/java/com/my/RoleCheckLink.java
package src.chainOfResponsibility.java.com.my;

public class RoleCheckLink extends ChainLink {
    @Override
    public boolean check(String username, String password, String token) {
        if (username.equals("admin")) {
            System.out.println("admin");
            return true;
        }

        System.out.println("user");
        return next(username, password, token);
    }
}
